src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_aad_auth.py

```diff
@@ -179,3 +222,115 @@ def test_azconfig_aad_auth(self, resource_group, location):
             exported_kvs = json.load(json_file)
         assert expected_exported_kvs == exported_kvs
         os.remove(exported_file_path)
+
+
+    @mock.patch('azure.cli.core.auth.adal_authentication.MSIAuthenticationWrapper.set_token')
+    @mock.patch('azure.cli.core.auth.adal_authentication.MSIAuthenticationWrapper.get_token')
+    @mock.patch('azure.cli.core._profile.Profile.get_subscription')
+    def test_azconfig_mi_token_override(self, get_subscriptions_mock, msi_get_token_mock, msi_set_token_mock):
+        get_subscriptions_mock.return_value = TEST_MI_SUBSCRIPTION
+        msi_get_token_mock.return_value = { "token": "mocked_token", "expires_on": 1234567890 }
+
+        mock_cmd = mock.MagicMock()
+        profile = Profile(cli_ctx=mock_cmd.cli_ctx)
+        cred, _, _ = profile.get_login_credentials()
+
+        appconfig_credential = AppConfigurationCliCredential(cred, APPCONFIG_AUTH_TOKEN_AUDIENCE)
+        
+        self.assertIsInstance(appconfig_credential._impl, MSIAuthenticationWrapper)
```

Comment:
Command modules should not assume the credential type. It is Azure CLI Core's internal implementation. Azure CLI Core only guarantees `get_token` is implemented.